reference.md

```diff
@@ -499,18 +540,124 @@ variables are missing:
 
 ``` fennel
 ;; bad
-(match [1 2 3]
+(case [1 2 3]
   ;; Will throw an error because `b' is nil for the first
   ;; pattern but the guard still uses it.
   (where (or [a 1 2] [a b 3]) (< a 0) (< b 1))
   :body)
 
 ;; ok
-(match [1 2 3]
+(case [1 2 3]
   (where (or [a b 2] [a b 3]) (< a 0) (<= b 1))
   :body)
 ```
 
+Symbols bound inside a `cases` pattern are independent from any existing
+symbols in the current scope, that is - names may be re-used without
+consequence.
+
+Example:
+
+```fennel
+(let [x 1]
+  (case [:hello]
+    ;; `x` is simply bound to the first value of [:hello]
+    [x] x)) ; -> :hello
+```
+
+Sometimes it may be preferable to match against an existing value in the outer
+scope. To do this we can "pin" a value from an existing symbol with a unary
+`=` expression. Pinning must be performed inside a `where` expression.
+
+Example:
+
+```fennel
+(let [x 1]
+  (case [1]
+    ;; 1 != :hello
+    (where [(= x)]) x
+    _ :no-match)) ; -> no-match
+
+(let [x 1]
+  (case [1]
+    ;; 1 == 1
+    (where [(= x)]) x
+    _ :no-match)) ; -> 1
+
+(let [pass :hunter2]
+  (case (user-input)
+    (where (= pass)) :login
+    _ :try-again!))
+```
+
+You must pin any symbol you wish to use in a where expression, but not in the
+pattern result expression.
+
+```fennel
+;; TODO make this a better example
+(let [limit 10]
+  (case [5 6]
+    (where [a b] (<= (= limit) (+ a b))) :over-the-limit
+    [a b] (+ a b limt)))
+```
+
+If a pattern defines a symbol with the same name as an existing symbol in the
+outer scope, the inner symbols value takes precedence.
+
+Example:
+
+```fennel
+(let [x 99]
+  (case [10 20]
+    ;; x is not pinned, so it's a new binding
+    [x y] (+ x y)) ; -> (+ 10 20)
+
+  (case [10 20]
+    ;; x was not rebound, so the existing symbol is used
+    [a b] (+ a x)) ; -> (+ 10 99)
+
+  (case [99 20]
+    ;; [99 20] = [99 b]
+    (where [(= x) b]) (+ x b)) ; -> (+ 99 20)
+
+  ;; PLEASE REVIEW
+  (case [99 20]
+    ;; [99 20] = [99 x]
+    (where [(= x) x]) (+ x x)) ; -> (+ 20 20)
```

Comment:
I think this is correct as the user has explicitly opted in to rebind x to something else, if they want to do `(+ 99 20)` then they should have used different symbol name!